functions/openhabian.bash

```diff
@@ -32,26 +31,52 @@ openhabian_console_check() {
     introtext="Additions, improvements or fixes were added to the openHABian configuration tool. Would you like to update now and benefit from them? The update will not automatically apply changes to your system.\\n\\nUpdating is recommended."
     if ! (whiptail --title "openHABian Update Available" --yes-button "Continue" --no-button "Skip" --yesno "$introtext" 15 80); then echo "SKIP"; return 0; fi
     echo ""
-    openhabian_update
+    openhabian_update "$branch"
   fi
+  echo -n "$(timestamp) [openHABian] Switching to branch $clonebranch ... "
+  # shellcheck disable=SC2015
+  git -C "$BASEDIR" checkout --quiet "$clonebranch" && echo "OK" || (FAILED=1; echo "FAILED"; return 0)
 }
 
 openhabian_update() {
+  local branch shorthash_before
+  current=$(git -C "$BASEDIR" rev-parse --abbrev-ref HEAD)
+  if [ "$current" == "master" ]; then
+    local introtext="You are currently using the very latest (\"master\") version of openHABian.\\nThis is providing you with the latest features but less people have tested it so it is a little more likely that you run into errors.\\nWould you like to step back a little now and switch to use the stable version ?\\nYou can switch at any time by selecting this menu option again or by setting the clonebranch= parameter in /etc/openhabian.conf.\\n"
+  else
+    if [ "$current" == "stable" ]; then
+      local introtext="You are currently using the stable version of openHABian.\\nAccess to the latest features would require you to switch to the latest version.\\nWould you like to step back a little now and switch to use the stable version ?\\nYou can switch versions at any time by selecting this menu option again or by setting the clonebranch= parameter in /etc/openhabian.conf.\\n"
+    else
+      local introtext="You are currently using neither the stable version nor the latest (\"master\") version of openHABian.\\nAccess to the latest features would require you to switch to master while the default is to use the stable version.\\nWould you like to step back a little now and switch to use the stable version ?\\nYou can switch versions at any time by selecting this menu option again or by setting the clonebranch= parameter in /etc/openhabian.conf.\\n"
+    fi
+  fi
+
   FAILED=0
+  if [[ -n "$INTERACTIVE" ]]; then
+    if [[ "$current" == "stable" || "$current" == "master" ]]; then
+      if ! sel=$(whiptail --title "openHABian version" --radiolist "$introtext" 14 75 2 stable "recommended standard version of openHABian" on master "very latest version of openHABian" off 3>&1 1>&2 2>&3); then return 0; fi
+    else
+      if ! sel=$(whiptail --title "openHABian version" --radiolist "$introtext" 14 75 3 stable "recommended standard version of openHABian" off master "very latest version of openHABian" off "$current" "some other version you fetched yourself" on 3>&1 1>&2 2>&3); then return 0; fi
+    fi
+    sed -i "s@^clonebranch=.*@clonebranch=$sel@g" "/etc/openhabian.conf"
+  fi
```

Comment:
So on patch day, are we dumping everyone straight to stable? If so, we are making them aware of the change, correct?

Reply:
Yes, we put them on stable.
Haven't decided yet how to make users aware of it.
I'm not even sure we should ?

Reply:
I think we should at least make a statement in the output, perhaps one that contains a link to a community post regarding the change and the new behavior of the utility. 

I don't think it is right to introduce a major change in the functionality / inner workings of the tool without making the user aware.

Reply:
Well it's not really a functionality change, is it ? Less than those we do every day ...
But you're right let's not annoy anybody by hiding that. I'll insert a popup.

Reply:
Like so ?

Reply:
Good enough, better than dealing with the unavoidable posts that say something is wrong.

Reply:
@ecdye @holgerfriedrich 
Just wondering, according to the install docs
https://www.openhab.org/docs/installation/openhabian.html#other-linux-systems-add-openhabian-just-like-any-other-software
if someone git cloned openhabian, he would be getting whatever branch is marked as 'default', wouldn't he ? That's currently 'master'.
Now as we want new users to get stable from now on, would we need to change the default branch ? I _guess_ (but do not know for sure) that that would have unwanted side effects such as whenever you create PRs they will (unless manually changed) also be against stable which also is not what we want.
So question: change default ? or change download instructions to include the stable branch ?

Reply:
Hmm, good question. Perhaps better to open a separate issue discuss it in.

Reply:
I added clone -b stable to the docs that's the safe route

Reply:
I noticed that just after commenting.

Reply:
You are correct in your assumption, GitHub would default to the stable branch if we marked that as the default branch. For now I think that the documentation change should be enough to avoid most of the issues with manual cloning.